### PR TITLE
Apply and conform to better linting

### PR DESCRIPTION
**Pull request by jwodder**
Includes the commit from #196.

This revealed a typing error which appears to be a genuine bug.

### CODE REVIEW

File: duecredit/io.py [hunk 1]
```diff
@@ -8,6 +8,8 @@
 # ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
 from __future__ import annotations
 
+from __future__ import annotations
```
Comment (a-detiste) [inline duecredit/io.py:11]: merge artifact, duplicate import

File: duecredit/io.py [hunk 2]
```diff
@@ -31,7 +33,7 @@
 from .versions import external_versions
 
 if TYPE_CHECKING:
-    from collector import Citation
+    from .collector import Citation
```
Comment (a-detiste) [inline duecredit/io.py:36]: thanks

File: duecredit/io.py (hunk 1 above)
Comment (jwodder) [reply duecredit/io.py]: Fixed.

**Comment by yarikoptic** [comment]
```
duecredit/io.py:396: error: Argument 1 to "get_bibtex_rendering" has incompatible type "DueCreditEntry"; expected "Union[Doi, BibTeX]"  [arg-type]
Found 1 error in 1 file (checked 50 source files)
```
Happy Thanksgiving!!!

**Comment by jwodder** [comment]
@yarikoptic Yes, that's the bug I found.  As far as I can tell, mypy is right to complain: the code is trying to pass a `DueCreditEntry` to a function that doesn't accept `DueCreditEntry`s, and I don't know what the intention behind the code is.

(The reason mypy is complaining now is because, after the import fix that @a-detiste commented on above, mypy now knows what a `Citation` is and what its `entry` attributes are.)

**Comment by yarikoptic** [comment]
for now I think the best would be to just make it accept `DueCreditEntry` as type... I also RFed the `get_text_rendering` to accept `DueCreditEntry` as well instead of `Citation`. Will rebase to resolve conflicts now

**Pull request merged**